client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -606,14 +606,15 @@ private ConcurrentHashMap<Integer, PartitionLocation> registerShuffleInternal(
         StatusCode respStatus = Utils.toStatusCode(response.getStatus());
         if (StatusCode.SUCCESS.equals(respStatus)) {
           ConcurrentHashMap<Integer, PartitionLocation> result = JavaUtils.newConcurrentHashMap();
-          for (int i = 0; i < response.getPartitionLocationsList().size(); i++) {
-            PartitionLocation partitionLoc =
-                PbSerDeUtils.fromPbPartitionLocation(response.getPartitionLocationsList().get(i));
-            pushExcludedWorkers.remove(partitionLoc.hostAndPushPort());
-            if (partitionLoc.hasPeer()) {
-              pushExcludedWorkers.remove(partitionLoc.getPeer().hostAndPushPort());
+          Tuple2<List<PartitionLocation>, List<PartitionLocation>> locations =
+              PbSerDeUtils.fromPbPackedPartitionLocationsPair(
+                  response.getPackedPartitionLocationsPair());
```

Comment:
Wont this not break when talking to older server ?

Reply:
This won't bother. Upgrade Celeborn should update the server first.

Reply:
In our internal deployment, we do ensure this - I was not sure how it gets used typically and if this was a scenario :)
Thanks for clarifying !